lib/app_profiler/backend.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module AppProfiler
+  class Backend
+    def run(params = {}, &block)
+      raise NotImplementedError
+    end
+
+    def start(params = {})
```

Comment:
I think this is in case someone ever wants to basically just directly profile something using a similar API to stackprof, ie, explicitly calling start and stop rather than using a block. And yeah, it is from before.